pkg/process/checks/discovery_check.go

```diff
@@ -0,0 +1,64 @@
+package checks
+
+import (
+	"os"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var ProcessDiscovery = &ProcessDiscoveryCheck{}
+
+type ProcessDiscoveryCheck struct {
+	probe *procutil.Probe
+	info  *model.SystemInfo
+}
+
+func (d *ProcessDiscoveryCheck) Init(cfg *config.AgentConfig, info *model.SystemInfo) {
+	d.probe = procutil.NewProcessProbe()
+	d.info = info
+}
+
+func (d *ProcessDiscoveryCheck) Name() string { return config.DiscoveryCheckName }
+
+func (d *ProcessDiscoveryCheck) RealTime() bool { return false }
+
+func (d *ProcessDiscoveryCheck) Run(_ *config.AgentConfig, groupID int32) ([]model.MessageBody, error) {
+	log.Info("Running process discovery check")
+	hostname, err := os.Hostname()
```

Comment:
We should follow the same approach to handle hostname as in other [checks](https://github.com/DataDog/datadog-agent/blob/main/pkg/process/checks/process.go#L183). It isn't necessarily the same as `os.Hostname()` - can be overriden by user, see [this](https://github.com/DataDog/datadog-agent/blob/main/pkg/process/config/config.go#L343-L350)

Reply:
Oh alright! Yeah that is a much better way to handle hostname